src/main/java/io/r2dbc/h2/H2ConnectionOption.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.r2dbc.h2;
+
+/**
+ * @author Greg Turnquist
+ */
+public enum H2ConnectionOption {
+
+    FILE_LOCK("FILE_LOCK"), // FILE|SOCKET|NO
```

Comment:
Is there a need for a key? I'd suggest to eliminate the `key` property and use `name()` within `getKey()`.

Also, we should migrate line comments to Javadoc-style, that way we get the documentation into Javadoc.

Reply:
Done.